testproxy/services/bulk-mutate-rows.js

```diff
@@ -14,38 +14,44 @@
 'use strict';
 
 const grpc = require('@grpc/grpc-js');
-const {BigtableClient} = require('../../build/src/index.js').v2;
 
 const normalizeCallback = require('./utils/normalize-callback.js');
+const getTableInfo = require('./utils/get-table-info');
 
 const bulkMutateRows = ({clientMap}) =>
   normalizeCallback(async rawRequest => {
     const {request} = rawRequest;
     const {request: mutateRequest} = request;
-    const {appProfileId, entries, tableName} = mutateRequest;
+    const {entries, tableName} = mutateRequest;
 
     const {clientId} = request;
     const bigtable = clientMap.get(clientId);
-    const client = new BigtableClient(bigtable.options.BigtableClient);
-    const result = await new Promise((res, rej) => {
-      const response = {entries: []};
-      client
-        .mutateRows({
-          appProfileId,
-          entries,
-          tableName,
-        })
-        .on('data', data => {
-          response.entries = [...data.entries];
-        })
-        .on('error', rej)
-        .on('end', () => res(response));
-    });
-
-    return {
-      status: {code: grpc.status.OK, details: []},
-      entry: result.entries,
-    };
+    const table = getTableInfo(bigtable, tableName);
+    try {
+      const mutateOptions = {
+        rawMutation: true,
+      };
+      await table.mutate(entries, mutateOptions);
+      return {
+        status: {code: grpc.status.OK, details: []},
+        entry: [],
+      };
+    } catch (error) {
+      if (error.name === 'PartialFailureError') {
+        return {
+          status: {code: grpc.status.OK, message: error.message},
+          entry: Array.from(error.errors.entries()).map(([index, entry]) => ({
+            index: index + 1,
+            status: entry,
+          })),
+        };
```

Comment:
why is the proxy turning partial failures into an "OK" response here?

Reply:
It meets the needs of the test case and is required for the test case to pass, but this is a good start to a discussion about what the test proxy *should* do.

Reply:
Maybe https://github.com/googleapis/nodejs-bigtable/blob/000a353f05fa65bc0a7dc551c777ac55c6c08875/src/table.ts#L1540 should change, but then we have a breaking change?

Reply:
We shouldn't change the node js client. 

You can return the status as is, just ensure the entries are filled correctly. I will update the test case as well as the guide of test proxy writeup.